### ISSUE TITLE
Alternative to using electron: Use "Progressive Web App" techniques

### ISSUE THREAD

**Issue by sjama** [TryGhost/Ghost-OSX#1]
What is a progressive web app? It's an app the progressively turns from a "website" into "app". 
This is mediated by the browser, provided the app meets certain conditions.

These are the steps as described by @slightlyoff [here](https://medium.com/@slightlylate/progressive-apps-escaping-tabs-without-losing-our-soul-3b93a8561955#.pp1nsv50i) 

1. The site begins life as a regular tab. It doesn’t have super-powers, but it is built using Progressive App features including TLS, Service Workers, Manifests, and Responsive Design.

2. The second (or third or fourth) time one visits the site — roughly at the point where the browser it sure it’s something you use frequently — a prompt is shown by the browser (populated from the Manifest details)

3. Users can decide to keep apps to the home screen or app launcher

4. When launched from the home screen, these apps blend into their environment; they’re top-level, full-screen, and work offline. Of course, they worked offline after step 1, but now the implicit contract of “appyness” makes that clear.

So rather then wrapping Ghost with electron and then distributing it as a downloadable "native" app, you can instead ensure sure Ghost meets the "Progressive Web App" requirements.

This has already been implemented on [Chrome Android](https://developers.google.com/web/updates/2014/11/Support-for-installable-web-apps-with-webapp-manifest-in-chrome-38-for-Android?hl=en) and there plans to bring it to other platforms.

The [Web App Manifest](https://w3c.github.io/manifest/) which power this is an open web standard, so you can expect other vendors to implement it in the future.

**Comment by JohnONolan** [comment]
@sjama this is a really interesting idea, but as far as I understand it - applies 100% to mobile only? This repo is solely about native desktop apps

**Comment by sjama** [comment]
@JohnONolan at the moment is available on Chrome Android and I think on Chrome OS as well. Because Google has more control over these platforms, they have implemented on them first.

But from what @slightlyoff told me in a recent meetup they are currently debating on the best way to bring it to other main Desktop platforms. 

On iOS they have much less control and the ball is Apple's court.

The main question is how important is Ghost Desktop app? Is it critical? or "a nice to have". 
If it's the former than going the native app route, is the way to go. 
Otherwise I would seriously consider waiting to see how "Progressive Web App" evolve over the next few months.

Because building and then maintaining a dedicated desktop app possibly for several platforms is not trivial task.

**Comment by JohnONolan** [comment]
This is all very interesting, but as you yourself have said - it's not widely adopted (it's barely adopted at all) - and at present it only even applies to Google mobile devices. So this is really not a suitable candidate for Ghost right now, nor is that going to change in the next few months.

I'm extremely aware of the work involved with building a cross-platform native app, that's kind of the entire point of us wanting to use Electron ;)

Closing this for now, but will certainly keep an eye on the progress of Progressive Web Apps in future.

**Comment by dandv** [comment]
Hey there, I'm Google Developer Advocate working on PWAs in @slightlyoff's larger team.

In the meantime, [PWAs have gained a lot of capabilities](https://medium.com/dev-channel/why-progressive-web-apps-vs-native-is-the-wrong-question-to-ask-fb8555addcbb), with [Twitter's PWA being the most recent high-profile launch](https://pinboard.in/u:dandv/b:7a6636fdb1ab).

We'd like to compile a list of feature needs that Electron satisfies which a PWA architecture still wouldn't, so as to motivate work towards better desktop support for PWAs.

**Comment by ColinEberhardt** [comment]
> @sjama this is a really interesting idea, but as far as I understand it - applies 100% to mobile only? This repo is solely about native desktop apps

Whilst PWAs were originally conceived as a way to deliver mobile apps outside of the 'stores' using web technologies, they are now very much being embraced on the desktop, with Microsoft supporting them in Edge early 2018:

https://www.thurrott.com/windows/windows-10/142649/microsofts-bold-plan-bring-pwas-windows-10

**Comment by slightlyoff** [comment]
We're starting work on Desktop PWAs in Chrome; also, Samsung has shipped Desktop PWA support for the DeX device. This is very much not desktop-only.

**Comment by JohnONolan** [comment]
This issue is so misguided it blows my mind that it's still going after almost 2 years.

We wrote the desktop app in electron because we — the maintainers, the people actually doing the work and writing the code — use electron, and we're very happy with how the app is working for us. 

I'm very sorry to break it to you guys, but what technology you happen to prefer is not remotely relevant to anything. You aren't even contributors to this repository.
